Feat/matches history page

zridi-mahmoud opened the pull request.

zridi-mahmoud committed:
feat(game history): added ranking icons for history table component

zridi-mahmoud committed:
feat(history table): added necessary components for table content with react table

zridi-mahmoud committed:
fix(games history): fix players card image rounding

zridi-mahmoud committed:
Merge branch 'main' into feat/games-history-table-component

zridi-mahmoud committed:
feat(history-table): improve UI and modularize column content

- Enhanced UI for better readability and user experience.
- Separated column content from main table component for better maintainability.
- Added `SortButton` component for sorting table in ascending, descending, or no specific order.

zridi-mahmoud committed:
feat(history-page): create the history page and useHistory hook

- create the history page with the layaout
- implement a useHistory to handle filteres and data fetching

zridi-mahmoud committed:
feat(history-page): complete UI, add responsiveness, filters, and debounce search

1. Completed the History table UI.
2. Added responsiveness to the History table.
3. Assembled the History page.
4. Added filters to the query parameters.
5. Added debounce to the search query.

zridi-mahmoud committed:
refactor(history-table): migrate from react-table to @tanstack/react-table v8

1. Updated table components to use @tanstack/react-table v8.
2. Refactored table configurations to align with the new API.
3. Updated dependencies in package.json.

zridi-mahmoud committed:
fix(history page): fix api route name and add limit to the params

zridi-mahmoud committed:
refactor(filename): rename useFilters.tsx to useGameHistory.tsx

zridi-mahmoud committed:
fix(history page): match UI design

zridi-mahmoud committed:
fix(pagination): fix pagination next button not disabled

zridi-mahmoud committed:
fix(history page): fix query debounce

oxidia commented on frontend/src/components/molecules/history/HistoryTableColumns.tsx:
```diff
@@ -0,0 +1,220 @@
+import { ColumnDef } from "@tanstack/react-table";
+import { Game } from "./HistoryTable";
+
+// eslint-disable-next-line no-shadow
```
You can remove this comment, I fixed it in another PR

oxidia commented on frontend/src/components/molecules/history/HistoryTableColumns.tsx:
```diff
@@ -0,0 +1,220 @@
+        return (
+          <div className="flex justify-center items-center gap-2 lg:gap-1 md:gap-1 sm:gap-1 xs:gap-1 sm:hidden">
+            <div className="text-light-fg-primary dark:text-dark-fg-primary">
+              {msToTime(duration.getTime())}
```
time format should be `mm:ss`

oxidia commented on frontend/src/app/history/[username]/page.tsx:
```diff
@@ -0,0 +1,142 @@
+
+const History = ({ token, username }: HistoryProps) => {
+  const { data, filters, setFilters } = useGameHistory(token, username);
+  const rowsPerPage = 4;
```
move this to useGameHistory hook and export it

// useGameHistory.ts
```ts
export const ROW_PER_PAGE = 4;
```

oxidia commented on frontend/src/components/molecules/history/useGameHistory.tsx:
```diff
@@ -0,0 +1,67 @@
+        params.append(key, value.toString());
+      }
+    });
+    params.append("limit", "10");
```
use the constant `ROW_PER_PAGE`

oxidia commented on frontend/src/components/molecules/feed/UserCard.tsx:
```diff
@@ -25,7 +25,7 @@ interface UserCardProps {
 
 const UserCard = (props: UserCardProps) => {
```
the rating diff must move a little bit to the right/left
![image](https://github.com/nerraou/ft_transcendence/assets/28894990/1c8156ca-0ea7-4fd6-ad28-d9d9b9251567)

zridi-mahmoud committed:
fix(history page): resolve multiple issues in table rendering and user handling

- Ensure table takes full height
- Handle user not found scenario
- Correct rows per page bug
- Correct query params keys
- Remove no-shadow rule disable comment
- Correct time format
- Increase datepicker's z-index for visibility

zridi-mahmoud committed:
fix(history page): color typo

zridi-mahmoud committed:
feat(history-page): Improved history page

- Change sorting mechanism to allow sorting by either time or date, but not both
- Fix content alignment issue in empty state, content should not be centered when there is no data
- Increase rounding on user card to match the UI design
